src/adagrams.js

```diff
@@ -1,15 +1,115 @@
+const LETTERPOOL = {
+  'A': 9, 
+  'B': 2, 
+  'C': 2, 
+  'D': 4, 
+  'E': 12, 
+  'F': 2, 
+  'G': 3, 
+  'H': 2, 
+  'I': 9, 
+  'J': 1, 
+  'K': 1, 
+  'L': 4, 
+  'M': 2, 
+  'N': 6, 
+  'O': 8, 
+  'P': 2, 
+  'Q': 1, 
+  'R': 6, 
+  'S': 4, 
+  'T': 6, 
+  'U': 4, 
+  'V': 2, 
+  'W': 2, 
+  'X': 1, 
+  'Y': 2, 
+  'Z': 1
+}
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+    let letterFreq = {};
+    let selectedLetter = [];
+    while ((selectedLetter.length) < 10) {
+      let random = Object.keys(LETTERPOOL)[Math.floor(Math.random()*Object.keys(LETTERPOOL).length)];
+          if (random in letterFreq) {
+              letterFreq[random] += 1;
+          } else {
+              letterFreq[random] = 1;
+          if (letterFreq[random] <= LETTERPOOL[random]){
+            selectedLetter.push(random)
+          }        
+          }        
+    }
+    return (selectedLetter)
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let lettersInHandCopy = lettersInHand
```

Comment:
Great idea but remember that this syntax won't make a copy of the list but rather a copy of the reference to the list. As a result, if you change lettersInHandCopy, you will also be changing lettersInHand. Research ways to make a deep copy instead to achieve what you were trying to achieve!